packages/vite-node/src/client.ts

```diff
@@ -129,7 +129,13 @@ export class ViteNodeRunner {
   }
 
   async executeFile(file: string) {
-    return await this.cachedRequest(`/@fs/${slash(resolve(file))}`, [])
+    try {
+      return await this.cachedRequest(`/@fs/${slash(resolve(file))}`, [])
+    }
+    catch (err) {
+      if (this.options.exitOnFailedTransform)
+        throw err
+    }
   }
```

Comment:
I think we should do the error handling on the `cli.ts` caller side instead

Reply:
@antfu Only getting back to this pr now, let me know if this works for you.